### PR TITLE
Enable new persistence handling for User & Role

### LINKED ISSUE #1301: using own user implementation
Hello dudes,

I created my own user implementation and extended it from sulu base user class:

```php
<?php

namespace Hautzi\Bundle\WebsiteBundle\Entity;

use Doctrine\ORM\Mapping AS ORM;
use Sulu\Bundle\SecurityBundle\Entity\User as BaseUser;

/**
 * @ORM\Entity
 */
class User extends BaseUser
{
    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

    /**
     * @ORM\OneToMany(targetEntity="Hautzi\Bundle\WebsiteBundle\Entity\TimelineItem", mappedBy="responsiblePerson")
     */
    private $timelineResponsibilites;

    /**
     * @ORM\OneToMany(targetEntity="Hautzi\Bundle\WebsiteBundle\Entity\TimelineItem", mappedBy="creator")
     */
    private $timelineItems;

    /**
     * Get id
     *
     * @return integer 
     */
    public function getId()
    {
        return $this->id;
    }
}

```

then I changed doctrine/orm/resolve_target_entities in `app/config/config.yml` to use this implementation

```php
        resolve_target_entities:
            Sulu\Component\Security\UserInterface: Hautzi\Bundle\WebsiteBundle\Entity\User
```

For relations everything works fine, but unfortunately when fetching a user from the Symfony security `TokenStorage`, I will receive the "wrong" Sulu User instance `Sulu\Bundle\SecurityBundle\Entity\User`.

Is there a way to configure an own user implementation that it is used across the whole application?

### PR DESCRIPTION
**Pull request by chirimoya**
https://github.com/sulu-io/sulu-standard/pull/470

__Tasks:__

- [x] test coverage
- [x] gather feedback for my changes
- [x] submit changes to the documentation
- [x] added changelog line

__Informations:__

| Q             | A
| ------------- | ---
| Tests pass?   | yes
| Fixed tickets | fixes #1301 
| BC Breaks     | none
| Doc           | https://github.com/sulu-io/sulu-docs/pull/44

### CODE REVIEW

File: src/Sulu/Bundle/SecurityBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -44,6 +45,40 @@ public function getConfigTreeBuilder()
...
+
         return $treeBuilder;
     }
+
+    /**
+     * Adds `objects` section.
+     *
+     * @param ArrayNodeDefinition $node
+     */
+    private function addObjectsSection(ArrayNodeDefinition $node)
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/DependencyInjection/Configuration.php]: @chirimoya i think this is mostly copied to all bundles? shouldnt we put this in a trait like `PersistenceConfigurationTrait `?

Comment (chirimoya) [inline src/Sulu/Bundle/SecurityBundle/DependencyInjection/Configuration.php]: No, this only separates the configuration and needs to be done for each bundle.

Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/DependencyInjection/Configuration.php]: i have created a issue for that maybe we can discuss this there: https://github.com/sulu-io/sulu/issues/1311

**Comment by wachterjohannes** [comment]
@chirimoya TOD (:

* Changelog
* Docs

**Comment by wachterjohannes** [comment]
Cookbock Entry: https://github.com/sulu-io/sulu-docs/pull/44

**Comment by danrot** [comment]
@chirimoya  Fixed the issue with setting the empty password on a put action.
It was actually not the empty password or the condition, the problem was that the salt was generated before the empty password check was applied. The tests were not failing because there is a plain password strategy used, so the salt was not applied at all. Don't know how to write a test for this...

**Comment by wachterjohannes** [comment]
ACK but i will wait for sulu-standard tests

**Comment by danrot** [comment]
sulu-standard probably won't work before this is merged

**Comment by wachterjohannes** [comment]
ah thats right (= i will merge both now we will see then!

**Pull request merged**